core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -1113,7 +1113,7 @@ private[spark] class MapOutputTrackerMaster(
       endMapIndex: Int): Seq[String] =
   {
     val shuffleStatus = shuffleStatuses.get(dep.shuffleId).orNull
-    if (shuffleStatus != null) {
+    if (shuffleStatus != null && shuffleLocalityEnabled) {
```

Comment:
nit: Why not avoid the `shuffleStatuses` query if `shuffleLocalityEnabled` is `false` and return `Nil` immediately ?

Reply:
Ok, I will change it.